### PR TITLE
(fixes #3464) Replace assert() usages with assertThat()

### LINKED ISSUE #3463: Remove usages of assert
```
/**
 * Throws an [AssertionError] if the [value] is false
 * and runtime assertions have been enabled on the JVM using the *-ea* JVM option.
 */
@kotlin.internal.InlineOnly
public inline fun assert(value: Boolean) {
    assert(value) { "Assertion failed" }
}
```
*and runtime assertions have been enabled on the JVM using the *-ea* JVM option.* 
They have not been enabled and won't be. Replace with appropriate `AssertThat`s.

### LINKED ISSUE #3464: App posts deletion request notifications ({{subst:idw}}) on the wrong user's talk page
I'm afraid I don't actually use the app: this issue is based on observing edits made on Commons by people apparently using it.

When someone nominates a file for deletion, a notification (using {{subst:[idw](https://commons.wikimedia.org/wiki/Template:Idw)}}) should be placed on the user talk page of everyone who uploaded a version of that file. It seems that instead, the app is placing such a notification on the user talk page of the user who nominated the image for deletion. This is a serious problem, since it means that uploaders may not find out that one of their uploads has been nominated for deletion.

Examples:
* [Commons:Deletion requests/File:The Valleys.png](https://commons.wikimedia.org/wiki/Commons:Deletion_requests/File:The_Valleys.png): file uploaded by Realitys; nominated by Magurale; notification appears on [User talk:Magurale](https://commons.wikimedia.org/wiki/User_talk:Magurale) but not on [User talk:Realitys](https://commons.wikimedia.org/wiki/User_talk:Realitys)
* [Commons:Deletion requests/File:Daegu1 Life Treatment Center (coronavirus)1.png](https://commons.wikimedia.org/wiki/Commons:Deletion_requests/File:Daegu1_Life_Treatment_Center_(coronavirus)1.png): file uploaded by Choikwangmo9; nominated by Kurtkaiser; notification appears on [User talk:Kurtkaiser](https://commons.wikimedia.org/wiki/User_talk:Kurtkaiser) but not on [User talk:Choikwangmo9](https://commons.wikimedia.org/wiki/User_talk:Choikwangmo9)
* [Commons:Deletion requests/File:Vilnia, Łukiški. Вільня, Лукішкі (V. Zacharčyk, 1865-79).jpg](https://commons.wikimedia.org/wiki/File:Vilnia,_%C5%81uki%C5%A1ki._%D0%92%D1%96%D0%BB%D1%8C%D0%BD%D1%8F,_%D0%9B%D1%83%D0%BA%D1%96%D1%88%D0%BA%D1%96_(V._Zachar%C4%8Dyk,_1865-79).jpg): file uploaded by Kazimier Lachnovič; nominated by Nitiphonbinsoh407; notification appears on [User talk:Nitiphonbinsoh407](https://commons.wikimedia.org/wiki/User_talk:Nitiphonbinsoh407) but not on [User talk:Kazimier Lachnovič](https://commons.wikimedia.org/wiki/User_talk:Kazimier_Lachnovi%C4%8D)
* [Commons:Deletion requests/File:中苏友谊纪念塔 2020.jpg](https://commons.wikimedia.org/wiki/Commons:Deletion_requests/File:%E4%B8%AD%E8%8B%8F%E5%8F%8B%E8%B0%8A%E7%BA%AA%E5%BF%B5%E5%A1%94_2020.jpg): file uploaded by Rowingbohe; nominated by Somesh Desai; notification appears on [User talk:Somesh Desai ](https://commons.wikimedia.org/wiki/User_talk:Somesh_Desai ) but not on [User talk:Rowingbohe](https://commons.wikimedia.org/wiki/User_talk:Rowingbohe)

### PR DESCRIPTION
**Pull request by u7683345**
Uses of the `assert()` function in Kotlin are inherited from Java and are only enabled in the JVM with the `-ea` flag (enable assertions). Since the `-ea` flag has _"not been enabled and won't be"_, these current `assert()` assertions don't do anything. 

To fix this  the `assertThat()` assertion is always active, regardless of JVM settings, so this would be more suitable.  

Fixes #3463

### **What changes did you make and why?**

- Replaced all occurrences of the `assert()` function to equivalent `assertThat()` function calls, in 3 relevant files.
- Imported 3 functions from the hamcrest.CoreMatchers library, `assetThat()` , `equalTo()`, and `not()`, in these files for translating the current assertions over to the hamcrest syntax.

### **Tests performed**

Tested `ProdDebug` on Pixel 9 with API level 34.

### CODE REVIEW

**Comment by u7683345** [comment]
@nicolas-raoul would you be able to review this? :-)

**Comment by nicolas-raoul** [comment]
Thanks!

Now I guess we have to find out why this test is failing:

```
fr.free.nrw.commons.location.LatLngTest > testConstructorBigLongitude FAILED
    java.lang.AssertionError at LatLngTest.kt:26
    
1542 tests completed, 1 failed, 40 skipped
```

File: app/src/test/kotlin/fr/free/nrw/commons/location/LatLngTest.kt [hunk 1]
```diff
@@ -14,51 +17,51 @@ class LatLngTest {
...
         latLng1 = LatLng(0.0, -181.0, 0.0f)
-        assert(latLng1.longitude == 179.0)
+        assertThat(latLng1.longitude, equalTo(179.0))
     }
 
     @Test
     fun testConstructorBigLongitude() {
         latLng1 = LatLng(0.0, 181.0, 0.0f)
-        assert(latLng1.longitude == -179.0)
+        assertThat(latLng1.longitude, equalTo(179.0))
```
Comment (nicolas-raoul) [inline app/src/test/kotlin/fr/free/nrw/commons/location/LatLngTest.kt:26]: Maybe missing a minus sign?

Comment (u7683345) [reply app/src/test/kotlin/fr/free/nrw/commons/location/LatLngTest.kt:26]: Ahh yep that would do it 😅

**Review by nicolas-raoul — APPROVED** [review]
Tests pass, thanks! :-)

**Pull request merged**